code/modules/power/singularity/field_generator.dm

```diff
@@ -317,7 +317,8 @@ no power level overlay is currently in the overlays list.
 			fields += CF
 			G.fields += CF
 			for(var/mob/living/L in T)
-				CF.Crossed(L)
+				var/turf/living_loc = get_turf(CF)
+				SEND_SIGNAL(living_loc, COMSIG_MOVABLE_CROSSED, L)
```

Comment:
More "signals should only be fired in one place". Containment fields should publicly expose a proc that murders the people on top of it, and you should just call that.

Reply:
just calls the on_crossed proc on the field now